### PR TITLE
modernize

### PR DESCRIPTION
**Pull request by RoadRunnr**
* bump minimum OTP to 21.0
* replace lager with logger
* replace obsolete functions

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,13 +7,9 @@ before_install:
...
 otp_release:
-    - 19.0
-    - 20.0
-    - 20.3.8.22
-    - 21.1
-    - 21.2.7
-    - 21.3.8.1
-    - 22.0.7
+    - 21.3
+    - 22.0
```
Comment (ebengt) [inline .travis.yml]: Is 22.0 okay? I thought you said latest 22.1?

File: rebar.config [hunk 2]
```diff
@@ -1,30 +1,35 @@
 %-*-Erlang-*-
 {erl_opts, [debug_info, {parse_transform, lager_transform}]}.
-{minimum_otp_vsn, "19.0"}.
+{minimum_otp_vsn, "21.0"}.
 
 {deps, [
-    {lager,          "3.7.0"},
-    {exometer_core,  "1.5.2"}
+    {exometer_core,  "1.5.7"}
```
Comment (ebengt) [inline rebar.config]: Will this be replaced with Prometheus?

File: README_eradius_journal.adoc [hunk 3]
```diff
@@ -1,5 +1,7 @@
 = Structured RADIUS Logging
 
+NOTE: This is outdated. Lager and it the lager journald backend was removed.
```
Comment (ebengt) [inline README_eradius_journal.adoc]: I think 'it' can be removed.

File: test/readme.md [hunk 4]
```diff
@@ -4,6 +4,8 @@ This directory is devoted for tests against the infrastructure of Eradius.
 
 ## Using the Logtest
 
+NOTE: This is outdated. Lager and it the lager journald backend was removed.
```
Comment (ebengt) [inline test/readme.md]: I think 'it' can be removed.

File: rebar.config (hunk 2 above)
Comment (mgumz) [reply rebar.config]: no, this change is just removing "lager"

File: test/eradius_client_SUITE.erl [hunk 5]
```diff
@@ -37,7 +37,7 @@ all() -> [
 init_per_suite(Config) ->
     {ok, _} = application:ensure_all_started(eradius),
     startSocketCounter(),
-    application:set_env(lager, handlers, [{lager_journald_backend, []}]),
+    %% application:set_env(lager, handlers, [{lager_journald_backend, []}]),
```
Comment (0xAX) [inline test/eradius_client_SUITE.erl:40]: Since lager was removed maybe we should remove this line as well instead of commenting it?

**Pull request merged**